app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineFragment.kt

```diff
@@ -83,7 +85,16 @@ class SearchEngineFragment : PreferenceFragmentCompat() {
         showSyncedTabsSuggestions.onPreferenceChangeListener = SharedPreferenceUpdater()
         showClipboardSuggestions.onPreferenceChangeListener = SharedPreferenceUpdater()
         searchSuggestionsInPrivatePreference.onPreferenceChangeListener = SharedPreferenceUpdater()
-        showVoiceSearchPreference.onPreferenceChangeListener = SharedPreferenceUpdater()
+        showVoiceSearchPreference.onPreferenceChangeListener = object : Preference.OnPreferenceChangeListener {
+            override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+                val newBooleanValue = newValue as? Boolean ?: return false
+                requireContext().settings().preferences.edit {
```

Comment:
`context` could also be returned by `preference.context` instead of `requireContext()`.

Reply:
Don't see any advantage in using one over the other.
Would prefer to stay with `requireContext()` just because it's already there.